File: online-judge-solutions/Codeforces/1539B.cpp
//https://codeforces.com/problemset/problem/1539/B
//Lang		:	GNU C++14
//Time		:	530 ms
//Memory	:	1200 KB
#include<iostream>
#include<vector>
using namespace std;
int main(){
	int n,q,l,r;
	while(cin>>n>>q){
		string s;
		cin>>s;
		vector<int> v;
		int suma=0;
		for(char x:s){
			suma+=x-96;
			v.push_back(suma);
		}
		v[-1]=0;
		while(q--){
			cin>>l>>r;
			cout<<v[r-1]-v[l-2]<<endl;
		}
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1539B.py
#https://codeforces.com/problemset/problem/1539/B
#Lang		:	Python 3.8
#Time		:	967 ms
#Memory         :	3400 KB
n,q=map(int,input().split())
s,suma,v=input(),0,[]
for i in range(len(s)):
    suma+=ord(s[i])-96
    v.append(suma)
for x in range(q):
    l,r=map(int,input().split())
    menos=0 if l-2<0 else v[l-2]
    print(v[r-1]-menos)
